### PR TITLE
C bogo sort

### PR DESCRIPTION
**Pull request by Gathros**

### CODE REVIEW

File: chapters/sorting_searching/bogo/code/c/bogo_sort.c [hunk 1]
```diff
@@ -0,0 +1,46 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+
+bool is_sorted(int *array, int n) {
```
Comment (ubsan) [inline chapters/sorting_searching/bogo/code/c/bogo_sort.c]: You should probably use `size_t` here?

File: chapters/sorting_searching/bogo/code/c/bogo_sort.c [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    while (--n >= 1) {
+        if (array[n] < array[n - 1]) {
+            return false;
+        }
+    }
+
+    return true;
+}
+
+void shuffle(int *array, int n) {
```
Comment (ubsan) [inline chapters/sorting_searching/bogo/code/c/bogo_sort.c]: again, probably use size_t?

File: chapters/sorting_searching/bogo/code/c/bogo_sort.c [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+void bogo_sort(int *array, int n) {
+    while (!is_sorted(array, n)) {
+        shuffle(array, n);
+    }
+}
+
+int main() {
+    int array[] = { 1, 3654, 78, 654, -234, -12, 4, 3, -6, -100};
```
Comment (ubsan) [inline chapters/sorting_searching/bogo/code/c/bogo_sort.c]: Probably should use a `#define`, and make certain that this array is of that size.

File: chapters/sorting_searching/bogo/code/c/bogo_sort.c (hunk 1 above)
Comment (Gathros) [reply chapters/sorting_searching/bogo/code/c/bogo_sort.c]: yeah, I should use `size_t`.

**Comment by leios** [comment]
alright, I think this one is good to merge. I realize we don't have a `#define`, but adding `array[10]` works as well here.

**Pull request merged**